src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +327,201 @@ else if (!period.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            if (isPeriodAtEnd(summaryVisible, period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
+            }
+            else if (containsForbiddenFragment(summaryVisible)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
+        }
+    }
+
+    /**
+     * Gets whole content of summary tag.
+     *
+     * @return Summary sentence of javadoc root node.
+     */
+    private String getInlineSummary() {
+        final DetailAST blockCommentAst = getBlockCommentAst();
+        return blockCommentAst.getFirstChild().getText();
+    }
+
+    /**
+     * Gets the string that is visible to user in javadoc.
+     *
+     * @param summary Whole content of summary javadoc.
+     * @return string that is visible to user in javadoc.
+     */
+    private static String getVisibleContent(String summary) {
+        final Matcher matcher = SUMMARY_PATTERN.matcher(summary);
+        String comment = "";
+        if (matcher.find()) {
+            comment = removeRedundantElementsFromString(matcher.group(1));
+        }
+        return comment.trim();
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
+        final String summarySentence = sentence.trim();
+        return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
+    }
+
+    /**
+     * Remove html tags from string.
+     * This is required as ANTLR does not parses html lists sometimes as HTML Elements.
+     * The issue link to this issue is
+     * <a href="https://github.com/checkstyle/checkstyle/issues/9703">Issue #9703</a>
+     *
+     * @param summarySentence string to clean.
+     * @return string without html tags.
+     */
+    private static String removeRedundantElementsFromString(String summarySentence) {
+        return REDUNDANT_ELEMENTS_PATTERN.matcher(summarySentence).replaceAll("");
+    }
+
+    /**
+     * Tests if first sentence contains forbidden summary fragment.
+     *
+     * @param firstSentence String with first sentence.
+     * @return true, if first sentence contains forbidden summary fragment.
+     */
+    private boolean containsForbiddenFragment(String firstSentence) {
+        final String javadocText = JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN
+                .matcher(firstSentence).replaceAll(" ").trim();
+        return forbiddenSummaryFragments.matcher(trimExcessWhitespaces(javadocText)).find();
+    }
+
+    /**
+     * Trims the given {@code text} of duplicate whitespaces.
+     *
+     * @param text The text to transform.
+     * @return The finalized form of the text.
```

Comment:
Here, return tag has capitalized first letter; in other places in this file, it does not. Also, `The` -> `the`.

Reply:
Let us stick to everything in small after param and return as shown in example in https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#examples

Reply:
Thanks for the link. I do not include a period at the end of the tag descriptions, since they usually aren't sentences, but this might be style preference.